src/infra/ingest-datafusion/src/writer.rs

```diff
@@ -851,16 +854,101 @@ impl DataWriterDataFusionBuilder {
             .resolve_ref(&self.block_ref)
             .await
             .int_err()?;
+        let mut seed_visitor = SearchSeedVisitor::create();
+        let mut set_vocab_visitor = SearchSetVocabVisitor::create();
+        let mut set_data_schema_visitor = SearchSetDataSchemaVisitor::create();
+        let mut prev_source_state_visitor = GenericCallbackVisitor::new(
+            None::<SourceState>,
```

Comment:
Searching for last source state I think deserves a dedicated visitor type.

This is actually a very interesting edge case that we will have to keep in mind for skip lists.